src/components/pages/Camera/VideoPreview.tsx

```diff
@@ -51,8 +53,31 @@ export class VideoPreview extends React.Component<
     };
   }
 
-  uploadVideo = async (videoUploadRef: firebase.storage.Reference) => {
-    const response = await fetch(this.props.videoUri);
+  compressAndUploadVideo = async () => {
+    const options = {
+      width: 480,
+      height: 640,
+      bitrateMultiplier: 3,
+      minimumBitrate: 300000
+    };
+    this.videoPlayerRef
```

Comment:
You can do this async await style

try {
  const newSource = await this.videoPlayerRef.compress(options);
  // bleh
catch (error) {
  // blah
}

Reply:
Am I supposed to mix try/catches with Promises? What do I return from error flow?

Reply:
in an `async` function, when a promise you `await` on results in an error in a `try` block, then it triggers the `catch` block; and when an `async` function throws an exception, it results in the Promise rejecting rather than doing a normal exception. 

So in fact it's better that way! the try/catch block will handle both normal exceptions and rejected promises, and throwing promises from an `async` function will do the expected behavior of rejecting the promise.